text/trace/0107-sampling-score.md

```diff
@@ -0,0 +1,217 @@
+   sampling rates across different components.
+
+## Explanation
+
+Sampling propabaility is generated by the first service to make sampling
+decision. It's a random float number in [0, 1] range.
+Score is stamped on the span and also propagated further within `tracestate`.
+
+Next service reads score from `tracestate` (instead of calculating it from
+trace-id) and compares it with its sampling rate to make sampling decision.
+
+Score is also exposed through span attributes. Vendors can leverage it
+to sort traces based on their completeness: the lower the value of score is,
+the higher the chance it was sampled it by each component.
+
+### Example
+
+```
++----------------------+     +----------------------+     +----------------------+
++ Service-A (rate 0.6) + --> + Service-B (rate 0.1) + --> + Service-C (rate 0.5) +
++-------------- -------+     +----------------------+     +----------------------+
+```
+
+1. Service-A receives a request
+   - starts a new trace, generates random trace-id
+   - generates score: `0.17935003`. It's **smaller** than sampling rate
+     (`0.6`), so decision is `RECORD_AND_SAMPLED`
+   - span gets a new attribute `sampling.score = 0.17935003`
+   - tracestate is modified `sampling.score=0.17935003`
+2. Service-B gets request from A
+   - reads trace-context from headers and `sampling.score` from the
+     tracestate
+   - decision is `NOT_RECORD` as `0.17935003` is **bigger** than its
+     sampling rate (0.1)
+3. Service-C get a request from B
+   - reads trace-context from headers and `sampling.score` from the
+     tracestate
+   - decision is `RECORD_AND_SAMPLED` as `0.17935003` is **smaller** than its
+     sampling rate (0.5)
+   - span gets a new attribute `sampling.score = 0.17935003`
+   - tracestate is left untouched
+
+As a result, spans from Service-A and Service-C are exported.
+It's not possible to restore relationship between A and C without B and the
+trace is broken, but Service-C can trace their own requests regardless of B's
+sampling rate and B can have smaller tracing budget regardless of A's decisions.
+All of them can still debug integration issues using common trace-id.
+
+Vendors can pick the most complete traces sorting them by score.
+
+## Internal details
+
+- Service that starts a trace makes sampling decision.  It's configured to use
+`ExternalScoreSampler`(name TBD) is configured by user. Within `ShouldSample`
+callback sampler
+  - generates random float score (6-9 digits) in [0, 1] interval
+  - makes sampling decision by comparing generated score to configured rate
+  - if decision is `RECORD` (or `RECORD_AND_SAMPLED`), sampler adds
+    `sampling.score` attribute to attributes collection of to-be-created span
+  - regardless of sampling decision: prepends `sampling.score` key-value pair
+    into tracestate of to-be-created span
+- Downstream service continues a trace but has different sampling rate (it's
+  also configured to use `ExternalScoreSampler`)
```

Comment:
What happens if not configured to use `ExternalScoreSampler`? I think I already asked this question, sorry for duplication.

Reply:
same as today - whatever sampler you configured would work according to the spec. Score (if present in tracestate) will be ignored and propagated futher